# **Healthcare Insurance Cost Analysis**

## Objectives

The objective of this notebook is to analyse a healthcare insurance dataset and understand how personal, lifestyle and geographic factors influence medical insurance charges.

This notebook will:
* Load and inspect the healthcare insurance dataset
* Check the dataset for missing values, duplicates and incorrect data types
* Clean and prepare the data for analysis
* Explore relationships between insurance charges and variables such as age, BMI, smoking status, sex, children and region
* Create visualisations to support the analysis
* Summarise the key findings, limitations and possible next steps

## Inputs

The input for this notebook is the `insurance.csv` dataset stored in the `data` folder.

The dataset contains the following columns:

* `age`: age of the insurance customer
* `sex`: gender of the customer
* `bmi`: body mass index
* `children`: number of children/dependants covered by insurance
* `smoker`: whether the customer is a smoker
* `region`: customer region
* `charges`: individual medical insurance charges

## Outputs

The outputs of this notebook will include:

* An inspected and cleaned version of the insurance dataset, if cleaning is required
* Summary statistics for the dataset
* Exploratory data analysis tables
* Visualisations showing patterns in insurance charges
* Written insights explaining which factors appear to influence insurance charges
* A final conclusion and suggested next steps

## Additional Comments

This notebook will be developed step by step, starting with loading and understanding the dataset before moving on to cleaning, exploratory data analysis, visualisations and conclusions.



---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'/Users/sahraosman/Documents/vscode-projects/Healthcare_Insurance/jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chdir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'/Users/sahraosman/Documents/vscode-projects/Healthcare_Insurance'

---

# Section 1: Load and Inspect the Dataset

In this section, the required Python libraries are imported for data analysis and visualisation. Pandas is then used to load the healthcare insurance dataset and inspect its initial structure.

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [8]:
df = pd.read_csv('data/insurance.csv')
df.head()

,age,sex,bmi,children,smoker,region,charges
0,19,female,27.900,0,yes,southwest,16884.92400
1,18,male,33.770,1,no,southeast,1725.55230
2,28,male,33.000,3,no,southeast,4449.46200
3,33,male,22.705,0,no,northwest,21984.47061
4,32,male,28.880,0,no,northwest,3866.85520


The first five rows show that the dataset has loaded successfully. It contains customer information including age, sex, BMI, number of children, smoking status, region and insurance charges.

---

# Section 2: Initial Data Inspection

In this section, the dataset will be inspected to understand its size, column names, data types and summary statistics.

In [10]:
df.shape

(1338, 7)

The dataset contains 1,338 rows and 7 columns. Each row represents one insurance customer, and each column represents a customer attribute or the insurance charge.

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


The dataset has no missing values shown in the non-null count, as each column contains 1,338 non-null entries. There are four numerical columns: `age`, `bmi`, `children` and `charges`, and three categorical columns: `sex`, `smoker` and `region`.

In [14]:
df.describe()

,age,bmi,children,charges
count,1338.000000,1338.000000,1338.000000,1338.000000
mean,39.207025,30.663397,1.094918,13270.422265
std,14.049960,6.098187,1.205493,12110.011237
min,18.000000,15.960000,0.000000,1121.873900
25%,27.000000,26.296250,0.000000,4740.287150
50%,39.000000,30.400000,1.000000,9382.033000
75%,51.000000,34.693750,2.000000,16639.912515
max,64.000000,53.130000,5.000000,63770.428010


The summary statistics show that customer ages range from 18 to 64, with an average age of approximately 39. The average BMI is approximately 30.66, and the number of children ranges from 0 to 5. Insurance charges vary widely, from around 1,121.87 to 63,770.43, which suggests that some customer characteristics may have a strong influence on charges.

In [15]:
df.columns

Index(['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'], dtype='object')

The column names are clear, consistent and lowercase. They do not contain spaces, so no column renaming is required at this stage.

---

# Section 3: Data Quality Checks

In this section, the dataset will be checked for duplicates, missing values, incorrect data types and any other issues that may require cleaning before analysis.

In [25]:
df.duplicated().sum()

1

This shows that there is 1 duplicate row in the dataset. The duplicate row will be removed to avoid counting the same record more than once in the analysis.

In [27]:
df.drop_duplicates(inplace=True)

This removes the duplicate row from the dataset, ensuring that each customer is only counted once in the analysis.

In [28]:
df.duplicated().sum()

0

This shows there are no duplicate rows in the dataset after removing the duplicate, as the `duplicated().sum()` method returns 0. Therefore, no further duplicate removal is required.

In [29]:
df.shape

(1337, 7)

After removing the duplicate row, the dataset now contains 1,337 rows and 7 columns.

In [20]:
df.isnull().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

This shows there are no missing values in the dataset, as the `isnull().sum()` method returns a Series of all zero values. Therefore, no missing value imputation is required.

In [26]:
df.dtypes

age           int64
sex          object
bmi         float64
children      int64
smoker       object
region       object
charges     float64
dtype: object

This shows the different data types in the dataset. The `age`, `bmi`, `children` and `charges` columns are numeric, while the `sex`, `smoker` and `region` columns are object types. These data types are appropriate for the initial analysis, so no data type conversion is required at this stage.

---